modular_skyrat/modules/advanced_shuttles/code/shuttles.dm

```diff
@@ -10,6 +14,55 @@
 
 	movement_force = list("KNOCKDOWN" = 3, "THROW" = 0)
 
+	///Our shuttle's control console
+	var/obj/machinery/computer/shuttle/arrivals/console
+	///How much time are we waiting before returning to interlink. Sets itself automatically from config file
+	var/wait_time
+	///State variable. True when our shuttle is waiting before autoreturn
+	var/waiting = FALSE // would've been better to use shuttle's mode variable, but check() resets it to SHUTTLE_IDLE so it's more sane way to make this fully modular
+
+/obj/docking_port/mobile/arrivals_skyrat/Initialize(mapload)
+	. = ..()
+	wait_time = CONFIG_GET(number/arrivals_wait)
+	return INITIALIZE_HINT_LATELOAD
+
+/obj/docking_port/mobile/arrivals_skyrat/LateInitialize()
+	. = ..()
+	console = get_control_console()
+
+/obj/docking_port/mobile/arrivals_skyrat/check()
+	. = ..()
+
+	if(!wait_time) //0 disables autoreturn
+		return
+	if(mode != SHUTTLE_IDLE)
+		return
+	if(waiting && !timer)
+		SSshuttle.moveShuttle(shuttle_id, ARRIVALS_INTERLINK, TRUE) // times up, we're leaving
+		waiting = FALSE
+	var/current_dock = getDockedId()
+	if (current_dock && current_dock == ARRIVALS_STATION)
+		for (var/P in GLOB.player_list)
+			var/mob/M = P
+			if(is_in_shuttle_bounds(M) && M.stat != DEAD)
+				if (waiting == TRUE)
+					timer = 0
+					waiting = FALSE
+					if(console && console.last_cancel_announce + CONSOLE_ANNOUNCE_COOLDOWN <= world.time)
+						console.say("Lifesigns detected onboard, automatic return aborted.")
+						console.last_cancel_announce = world.time
+				return
+		if (!timer && !waiting)
+			setTimer(wait_time)
+			waiting = TRUE
+			if(console && console.last_depart_announce + CONSOLE_ANNOUNCE_COOLDOWN <= world.time)
+				console.say("Commencing automatic return subroutine in [wait_time / 10] seconds.")
+				console.last_depart_announce = world.time
```

Comment:
I've been trying to rewrite this a bit as a suggestion, and after looking into the code some more, I think the best way to go about this would be to copy some of the behavior from `/obj/docking_port/mobile/arrivals`, namely the `PersonCheck()` and the `NukeCheck()` procs (properly renamed to use `snake_case` instead), as well as the `areas` variable. This way, it would allow you to do this a little more cleanly, especially because I'm not sure that using `is_shuttle_in_bounds()` is the best way to go about doing this.

If you need some help with that, you can poke me on Discord, my username is `goldenalpharex`.

Reply:
Done. I think. Still don't like how it looks, but i'm not a programmer and i just dont like how nested it is. Using `areas` is definitely better (and turns out, we already have this list stored for every shuttle) if someone decides to make crescent, donut or whatever shaped arrivals shuttle. I haven't thought of it when made this, just decided to use simple `coords inside a rectangle` path.

Reply:
also, sorry for delaying, i am just a lazy ass lizard